graphql/appointment/fields.ts

```diff
@@ -0,0 +1,188 @@
+    firstname: string;
+    @Field((_type) => String, { nullable: true })
+    lastname: string;
+    @Field((_type) => Boolean, { nullable: true })
+    isStudent?: boolean;
+    @Field((_type) => Boolean, { nullable: true })
+    isPupil?: boolean;
+    @Field((_type) => Boolean, { nullable: true })
+    isScreener?: boolean;
+}
+
+@ObjectType()
+class Organizer {
+    @Field((_type) => String, { nullable: true })
+    userId: string;
+    @Field((_type) => Int)
+    id: number;
+    @Field((_type) => String)
+    firstname: string;
+    @Field((_type) => String)
+    lastname: string;
+    @Field((_type) => Boolean)
+    isStudent: boolean;
+}
+
+@Resolver((of) => Appointment)
+export class ExtendedFieldsLectureResolver {
+    @Query((returns) => Appointment)
+    @AuthorizedDeferred(Role.OWNER, Role.APPOINTMENT_PARTICIPANT)
+    async appointment(@Ctx() context: GraphQLContext, @Arg('appointmentId') appointmentId: number) {
+        const appointment = await prisma.lecture.findUniqueOrThrow({ where: { id: appointmentId } });
+        await hasAccess(context, 'Lecture', appointment);
+        return appointment;
+    }
+
+    @Deprecated('use isOrganizer instead')
+    @FieldResolver((returns) => Boolean)
+    @Authorized(Role.ADMIN, Role.STUDENT)
+    async isInstructor(@Ctx() context: GraphQLContext, @Root() appointment: Appointment, @Arg('studentId', { nullable: true }) studentId?: number) {
+        const student = await getSessionStudent(context, studentId);
+        return appointment.instructorId === student.id;
+    }
+    @FieldResolver((returns) => Boolean)
+    @Authorized(Role.UNAUTHENTICATED)
+    async isOrganizer(@Ctx() context: GraphQLContext, @Root() appointment: Appointment, @Arg('studentId', { nullable: true }) studentId?: number) {
+        if (!isElevated(context) && !isSessionStudent(context)) {
+            return false;
+        }
+        const isOrganizer = (await prisma.lecture.count({ where: { id: appointment.id, organizerIds: { has: context.user.userID } } })) > 0;
+        return isOrganizer;
+    }
+
+    @FieldResolver((returns) => Boolean)
+    @Authorized(Role.USER)
+    async isParticipant(@Ctx() context: GraphQLContext, @Root() appointment: Appointment) {
+        const user = await getUserForSession(context.sessionToken);
+        const isParticipant = (await prisma.lecture.count({ where: { id: appointment.id, participantIds: { has: user.userID } } })) > 0;
+        return isParticipant;
+    }
+    @FieldResolver((returns) => [AppointmentParticipant], { nullable: true })
+    @Authorized(Role.OWNER, Role.APPOINTMENT_PARTICIPANT)
+    @LimitEstimated(30)
+    async participants(@Root() appointment: Appointment, @Arg('take', (type) => Int) take: number, @Arg('skip', (type) => Int) skip: number) {
+        const studentIds = [];
+        const pupilIds = [];
+        appointment.participantIds.forEach((userId) => {
+            const [type, id] = getUserTypeAndIdForUserId(userId);
```

Comment:
Could we move this whole thing to common/user.ts? Participant and Organizer should both be subsets of User, so we could just do `getUsers(userIds).map(it => { ...subset })` here